Release notes for #15614: Allow World::entity family of functions to take multiple entities and get multiple references back (#1825)
bevyengine/bevy-website: release-content/0.15/release-notes/15614_Allow_Worldentity_family_of_functions_to_take_multiple_ent.md

```rust
#[derive(Component)]
#[component(on_add = on_foo_added)]
struct Foo;

fn on_foo_added(world: DeferredWorld, entity: Entity, _: ComponentId) {
    let has_foo = world.entity(entity);
    println!("{:?} has a Foo", has_foo.id());
}
```

In previous versions of Bevy, you could grab multiple entities from a `World` using a variety of different functions:

- `World::many_entities<N>(&self, [Entity; N]) -> [EntityRef; N]`
- `World::many_entities_mut<N>(&mut self, [Entity; N]) -> [EntityMut; N]`
- `World::get_many_entities<N>(&self, [Entity; N]) -> Result<[EntityRef; N], Entity>`
- `World::get_many_entities_dynamic(&self, &[Entity]) -> Result<Vec<EntityRef>, Entity>`
- `World::get_many_entities_mut<N>(&mut self, [Entity; N]) -> Result<[EntityMut; N], QueryEntityError>`
- `World::get_many_entities_dynamic_mut(&self, &[Entity]) -> Result<Vec<EntityMut>, QueryEntityError>`
- `World::get_many_entities_from_set_mut(&mut self, &EntityHashSet) -> Result<Vec<EntityMut>, QueryEntityError>`

As you can see, that's a lot of functions with very long names! But the gist of them is that we want to support the ability to

> give a bunch of entity IDs, and receive a bunch of entity references. **We can do better.**

Reviewer: ```suggestion
give a bunch of entity IDs, and receive a bunch of entity references. Surely there's a better way!
```

> So in `0.15`, all of those functions have been deprecated and now all you need is

Reviewer: ```suggestion
In `0.15`, all of those functions have been deprecated and now all you need is
```

> So in `0.15`, all of those functions have been deprecated and now all you need is

Reviewer: This should include a quick link to the trait that enabled this for curious readers to see how this was done.

Author: I've written a good starting point, but it can probably be written out better.
